lib/SILOptimizer/Utils/Existential.cpp

```diff
@@ -207,11 +205,16 @@ OpenedArchetypeInfo::OpenedArchetypeInfo(Operand &use) {
     // Handle:
     //   %opened = open_existential_addr
     //   %instance = alloc $opened
-    //   copy_addr %opened to %stack
+    //   <copy|store> %opened to %stack
     //   <opened_use> %instance
-    if (auto stackInitVal =
-            getAddressOfStackInit(instance, user, isOpenedValueCopied)) {
-      openedVal = stackInitVal;
+    if (auto *initI = getStackInitInst(instance, user, isOpenedValueCopied)) {
+      if (auto *IEA = dyn_cast<InitExistentialAddrInst>(initI))
+        // TODO: this case doesn't need to exist beacuse openedVal will never be used
+        openedVal = IEA;
```

Comment:
@atrick any reason I shouldn't remove this if-statement?

Reply:
I would replace this with a comment that the init_existential_addr case is not handled here because it isn't considered an "opened" archetype, and that case should be handled by ConcreteExistentialInfo instead.